contracts/utils/KlimaCarbonRetirements.sol

```diff
@@ -0,0 +1,287 @@
+    }
+
+    /**
+        @notice This function updates the total claimed amount for minting an NFT.
+        @param _minter Address of the user trying to mint.
+        @param _amount Amount being claimed for the mint. Expected value in 18 decimals.
+     */
+    function offsetClaimed(address _minter, uint256 _amount)
+        public
+        returns (bool)
+    {
+        require(
+            isMinterContract[_minter],
+            "Calling contract is not an approved minter."
+        );
+        require(
+            getUnclaimedTotal(_minter) >= _amount,
+            "Trying to claim too many offsets"
+        );
+
+        Retirement storage info = retirements[_minter];
+        info.totalClaimedNFT += _amount;
+        return true;
+    }
+
+    /**
+        @notice This returns information on a specific retirement for an address.
+        @param _retiree Address that retired the offsets.
+        @param _index Index of all retirements made. Starts at 0.
+        @return Returns a tuple of the address for the pool address, amount offset in 18 decimals, and beneficiary description used in the retirement.
+     */
+    function getRetirementIndexInfo(address _retiree, uint256 _index)
+        public
+        view
+        returns (
+            address,
+            uint256,
+            string memory
+        )
+    {
+        return (
+            retirements[_retiree].retiredPool[_index],
+            retirements[_retiree].retiredAmount[_index],
+            retirements[_retiree].retirementBeneficiary[_index]
+        );
+    }
+
+    /**
+        @notice This returns the total amount offset by an address for a specific pool.
+        @param _retiree Address that performed the retirement.
+        @param _pool Address of the pool token.
+        @return Int with 18 decimals for the total amount offset for this pool token.
+     */
+    function getRetirementPoolInfo(address _retiree, address _pool)
+        public
+        view
+        returns (uint256)
+    {
+        return retirements[_retiree].totalPoolRetired[_pool];
+    }
+
+    /**
+        @notice This returns totals about retirements and claims on an address
+        @param _retiree Address that performed the retirement.
+        @return Int tuple. Total retirements, total tons retired, total tons claimed for NFTs.
+     */
+    function getRetirementTotals(address _retiree)
+        public
+        view
+        returns (
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return (
+            retirements[_retiree].totalRetirements,
+            retirements[_retiree].totalCarbonRetired,
+            retirements[_retiree].totalClaimedNFT
+        );
+    }
+
+    /**
+        @notice Allow contract owner to whitelist new helper contracts. This is to prevent writing abuse from external interfaces.
+        @param _helper Address of the helper contract.
+     */
+    function addHelperContract(address _helper)
```

Comment:
Maybe adding a `removeHelperContract` could be useful as well?
Also I would add a check if `_helper != address(0)`